### PR TITLE
[ARC-613] Image security scan

### PR DESCRIPTION
**Pull request by ewaldschrader**
branch: feature/ARC-613-image-scan-trivy

- add scan stage
- echo scan step
- add docker scan step

### CODE REVIEW

File: src/mpyl/steps/__init__.py [hunk 1]
```diff
@@ -105,13 +105,13 @@ class Step(metaclass=IPluginRegistry):
     """Is set to something other than `ArtifactType.NONE` if this step depends on an artifact produced by the execution
     of an earlier step. For example: a step in the `Deploy` stage, may need to deploy a docker image that was produced
     in the `Build` stage."""
-    before: Optional[Step]
-    after: Optional[Step]
+    before: Optional[list[Step]]
+    after: Optional[list[Step]]
```
Comment (koeves) [inline src/mpyl/steps/__init__.py:109]: Wondering if an optional list makes sense.. Is there a difference between a `None` and an empty list? If there're no steps before or after, should be an empty list I think

File: src/mpyl/steps/steps.py [hunk 2]
```diff
@@ -125,19 +125,22 @@ def _execute_stage(self, stage: Stage, project: Project, dry_run: bool = False)
                 artifact: Optional[Artifact] = self._find_required_artifact(project, executor.required_artifact)
                 result = Output(success=True, message='')
                 if executor.before:
-                    result = self._execute(executor.before, project, self._properties,
-                                           self._find_required_artifact(project, executor.before.required_artifact),
-                                           dry_run)
+                    for before_step in executor.before:
+                        result = self._execute(before_step, project, self._properties,
+                                               self._find_required_artifact(project, before_step.required_artifact),
+                                               dry_run)
```
Comment (koeves) [inline src/mpyl/steps/steps.py:131]: We should probably combine the results here then. Right now it just takes the last execution

File: src/mpyl/steps/test/dockertest.py [hunk 3]
```diff
@@ -21,8 +22,8 @@ def __init__(self, logger: Logger) -> None:
             logger=logger, meta=meta,
             produced_artifact=ArtifactType.JUNIT_TESTS,
             required_artifact=ArtifactType.NONE,
-            before=IntegrationTestBefore(logger),
-            after=IntegrationTestAfter(logger)
+            before=[IntegrationTestBefore(logger)],
+            after=[ScanDocker(logger), IntegrationTestAfter(logger)]
```
Comment (koeves) [inline src/mpyl/steps/test/dockertest.py:26]: Can we do any assertion that this after step succeeded? Or it triggers automatically for the sample projects and that's enough?

**Review by koeves — CHANGES_REQUESTED** [review]
Some smaller things to check

File: src/mpyl/steps/__init__.py (hunk 1 above)
Comment (ewaldschrader) [reply src/mpyl/steps/__init__.py:109]: Discussed with Sam, will revert this and add a `Scan Stage` instead.

**Pull request closed without merging**